packages/expo-router/src/modal/Modal.tsx

```diff
@@ -64,6 +64,13 @@ export interface ModalProps extends ViewProps {
    * while **Android is limited to three**.
    */
   detents?: ModalConfig['detents'];
+  /**
+   * See {@link ScreenProps["initialDetentIndex"]}.
+   *
+   * The initial detent index when sheet is presented.
+   * Works only when `presentation` is set to `formSheet`.
+   */
+  detentIndex?: ModalConfig['detentIndex'];
```

Comment:
I think that we should keep the `initialDetentIndex` as it suggests that it is initial, rather then can be changed when modal is open